pkgs/by-name/fi/firefoxpwa/package.nix

```
@@ -16,84 +16,111 @@
+      chmod -R +w $out/share/firefoxpwa
 
-    # UserChrome
-    cp -r userchrome $out/share/firefoxpwa
+      # UserChrome
+      cp -r userchrome $out/share/firefoxpwa
 
-    # Runtime patching
-    FFPWA_USERDATA=$out/share/firefoxpwa $out/bin/firefoxpwa runtime patch
+      # Runtime patching
+      FFPWA_USERDATA=$out/share/firefoxpwa $out/bin/firefoxpwa runtime patch
 
-    # Manifest
-    sed -i "s!/usr/libexec!$out/bin!" manifests/linux.json
-    install -Dm644 manifests/linux.json $out/lib/mozilla/native-messaging-hosts/firefoxpwa.json
+      # Manifest
+      sed -i "s!/usr/libexec!$out/bin!" manifests/linux.json
+      install -Dm644 manifests/linux.json $out/lib/mozilla/native-messaging-hosts/firefoxpwa.json
 
-    installShellCompletion --cmd firefoxpwa \
-      --bash $completions/firefoxpwa.bash \
-      --fish $completions/firefoxpwa.fish \
-      --zsh $completions/_firefoxpwa
+      installShellCompletion --cmd firefoxpwa \
+        --bash $completions/firefoxpwa.bash \
+        --fish $completions/firefoxpwa.fish \
+        --zsh $completions/_firefoxpwa
 
-    # AppStream Metadata
-    install -Dm644 packages/appstream/si.filips.FirefoxPWA.metainfo.xml $out/share/metainfo/si.filips.FirefoxPWA.metainfo.xml
-    install -Dm644 packages/appstream/si.filips.FirefoxPWA.svg $out/share/icons/hicolor/scalable/apps/si.filips.FirefoxPWA.svg
+      # AppStream Metadata
+      install -Dm644 packages/appstream/si.filips.FirefoxPWA.metainfo.xml $out/share/metainfo/si.filips.FirefoxPWA.metainfo.xml
+      install -Dm644 packages/appstream/si.filips.FirefoxPWA.svg $out/share/icons/hicolor/scalable/apps/si.filips.FirefoxPWA.svg
 
-    wrapProgram $out/bin/firefoxpwa \
-      --prefix FFPWA_SYSDATA : "$out/share/firefoxpwa" \
-      --prefix LD_LIBRARY_PATH : "$libs" \
-      --suffix-each GTK_PATH : "$gtk_modules"
+      wrapProgram $out/bin/firefoxpwa \
+        --prefix FFPWA_SYSDATA : "$out/share/firefoxpwa" \
+        --prefix LD_LIBRARY_PATH : "$libs" \
+        --suffix-each GTK_PATH : "$gtk_modules"
 
-    wrapProgram $out/bin/firefoxpwa-connector \
-      --prefix FFPWA_SYSDATA : "$out/share/firefoxpwa" \
-      --prefix LD_LIBRARY_PATH : "$libs" \
-      --suffix-each GTK_PATH : "$gtk_modules"
-  '';
+      wrapProgram $out/bin/firefoxpwa-connector \
+        --prefix FFPWA_SYSDATA : "$out/share/firefoxpwa" \
+        --prefix LD_LIBRARY_PATH : "$libs" \
+        --suffix-each GTK_PATH : "$gtk_modules"
+    '';
 
-  passthru.tests.firefoxpwa = nixosTests.firefoxpwa;
+    postInstallCheck = ''
+      mv $out/bin/firefoxpwa $out/bin/.firefoxpwa-wrapped
+    '';
+  };
+
+  # firefoxpwa wants to run binaries downloaded into users' home dir
+  fhs = buildFHSUserEnv {
+    name = "${name}-wrapper-${version}";
+    runScript = "${unwrapped}/bin/.firefoxpwa-wrapped";
+    targetPkgs = pkgs: [
+      pkgs.dbus-glib
+      pkgs.gtk3
+      pkgs.alsaLib
+      pkgs.xorg.libXtst
+      pkgs.xorg.libX11
+    ];
```

Perhaps use firefox-unwrapped's buildInputs?

```suggestion
    targetPkgs = _: firefox-unwrapped.buildInputs;
```